src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+// - if player "UNO"
+// - card drawed from the deck
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include "cards_management.h"
+ 
+#define MAX 40
+#define hand_MAX 20
+#define plarCardNumber 10
+
+///* (not used for now) */
+//struct CARD record_array[MAX]; // array for recording all the cards
+//int record_size = 0, handcard_size = 0;
+//
+//void record(struct CARD card)
+//{
+//    record_size++;
+//    record_array[record_size - 1] = card;
+//}
+
+/* 
+* @brief: function that find the most occurrence with the color 
+* 
+* @return CardColor the color with moset occurrence
+*/
+enum CardColor  find_most_color(struct DECK *cards_on_hand)
+{
+    int R_count = 0, G_count = 0, B_count = 0, Y_count = 0, most = 0, least = 0;
+    struct DECK *temp;
+    struct DECK *head = cards_on_hand;
+    if (head == NULL)
+    {
+        printf("hand card is empty!");
+        return 0;
+    }
+    temp = head;
+    while (temp != NULL)
+    {
+        if (temp->card.color == RED)
+        {
+            R_count++;
+        }
+        else if (temp->card.color == BLUE)
+        {
+            B_count++;
+        }
+        else if (temp->card.color == GREEN)
+        {
+            G_count++;
+        }
+        else
+        {
+            Y_count++;
+        }
+        temp = temp->next;
+    }
+    if (R_count > B_count && R_count > G_count && R_count > Y_count)
+    {
+        return RED;
+    }
+    else if (B_count > R_count && B_count > G_count && B_count > Y_count)
+    {
+        return BLUE;
+    }
+    else if (G_count > R_count && G_count > B_count && G_count > Y_count)
+    {
+        return GREEN;
+    }
+    else if (Y_count > R_count && Y_count > B_count && Y_count > G_count)
+    {
+        return YELLOW;
+    }
+
+    return 0;
+}
+
+/* function that choose the right card to draw */
+/*
+* @brief: function that choose the right card to draw
+*          if player card # > 1
+*           case 1 : no matched color and no matched number
+*             - if have wild card, draw wild if wild card > with color with the most occurence
+*             - if have skip, use skip
+*             - if not have wild or skip, take card from pile
+*            case 2 : have the matched color and no matched number
+*             - choose the largest number on that color
+*             - if matched color > 3, use match draw/draw2/skip card based on the color with the most occurence
+*
+*            case 3 : no matched color, but have matched number
+*             - if have multiple, choose the color with the most occurence and corresponding action card
+*             - if have only one, get rid of it
+*
+*            case 4 : have matched color and match number
+*             - quickly get rid of matched numbers in the hand first.
+*             - if occurence of
+*
+*         if player card # = 1
+*            - use action card if possibile
+*            - avoid play the last color that the player played
+*
```

Comment:
Same comments as above: Let's add the description of input parameter here.
Please check all the function descriptions as well.